### ISSUE TITLE
Apparent framerate drop.

### ISSUE THREAD

**Issue by mirage335** [baffler/OBS-OpenVR-Input-Plugin#26]
After this had been working for me just fine for a while, it seems the SteamVR mirror window now shows a drastically lower framerate.

Consequently, OBS recordings show VR footage at very low framerates... 10FPS if that.

**Comment by blackote** [comment]
Im having the same issues... mine looks like 25fps..  I noticed that in the SteamVR loading map (where all you see is the floor grid and mountains) the FPS is fine.  But once it load into the SteamVR home its choppy from that point on.

**Comment by BOLL7708** [comment]
We did some exploration [url=https://steamcommunity.com/app/250820/discussions/3/2146470329514484710/]here[/url], it appears it's due to the mix of WMR in combination with SteamVR that does it. Or are any of you using a non-WMR headset?

**Comment by mirage335** [comment]
Myself, I am using a Pimax Vision _8kX_ (preproduction unit). This obstructed better quality footage for the video I was producing...

**Comment by jackofchaos** [comment]
This is a problem for oculus as well

**Comment by Da-Nuker** [comment]
There was a test done with a rollback version of SteamVR over at that shows OpenVR 1.4 plugin working correctly on the 1.10.20 SteamVR for WMR: https://steamcommunity.com/app/250820/discussions/0/2268069450214724270/?ctp=2

So the bug maybe not be related to the plugin or unless there is a way to rewrite the plugin so that it can grab higher native output from SteamVR itself.

**Comment by mirage335** [comment]
It is a blocking issue for using the plugin's core functionality, so SteamVR being at fault should not be a reason to close the issue.

**Comment by Da-Nuker** [comment]
> It is a blocking issue for using the plugin's core functionality, so SteamVR being at fault should not be a reason to close the issue.

I agree leaving this issue here so people will know about it (and how to roll back to a working version), but I think it would also help more to keep reminding Valve's SteamVR team to fix this on their end.

Looking through the source, this plugin is mainly processing a copy of the mirror surface provided by the SteamVR Compositor and I'm not sure what else can be done without access into that unless they make you buy Valve Index to fix a recording problem they caused...

**Comment by mirage335** [comment]
How? How can we 'keep reminding', or something stronger? This kind of regression being left open for weeks is _absurd_.

Having a Valve Index _DOES NOT_ solve the problem. A Valve Index does _NOT_ help me record footage from the _Pimax Vision 8kX_, which is much higher resolution.

Personally, I am very fed up with the apparent mentality behind the supposedly 'OpenVR'.

**Comment by blackote** [comment]
I just tried the downgrade bandaid and it worked.... hopefully with the new win10 may update and more steamVR updates it will fix it.

**Comment by mirage335** [comment]
Another 20 days.

**Comment by BOLL7708** [comment]
Pretty sure you're barking up the wrong tree here @mirage335 as it appears to be a change in SteamVR that has caused this and not this specific plugin. It would be hard for people external to Valve to actually fix this unless it's up to driver makers to do some update, meaning Pimax, WMR, etc, which still means it's not a fix to this plugin 🤷‍♂️

**Comment by mirage335** [comment]
Pardon me. In fact, I had recently confirmed the plugin still did not work with recent versions of SteamVR stable and beta. So, I thought I might drop a note here to record the length of time this plugin's core functionality has remained blocked by SteamVR.

My 'barking' is already directed up different trees.

That said, might I propose it is now time to include a workaround in this repository? Something automatic, like a downgrade script that does not depend on specific memory editor programs that might be mistaken for cheats?

**Comment by Da-Nuker** [comment]
The rollback instructions are located on SteamVR discussion boards that I have posted. Here's the direct post: https://steamcommunity.com/app/250820/discussions/0/2268069450214724270/#c2268069450221705718

This will bring you back to 1.10.20 which is definitely working with this plugin to get smooth 90 fps vs the SteamVR desktop mirror itself which still only gives you half your refresh rate (45 fps) on this version. I believe 1.10.22 hotfix broke it since it had this in the changelog: "**Possible fix for crashes on Desktop View shutdown.**"

Their possible "fix" basically messed up VR capture to 30 fps for every non 1st-party SteamVR device.  Right now we are still all right as most software still runs on 1.10 but we eventually get some that needs 1.11+ features.

**Comment by mirage335** [comment]
Thank you for identifying the specific hotfix that is plausibly implifcated.

As for the workaround, it's far from ideal. Especially since it begins with this.
"1. Run Cheat Engine."

Seems like a great way to get a VAC permaban.

**Comment by Da-Nuker** [comment]
That's something you have to blame Valve for not blocking console commands to download repository which is a separate issue. They blocked it a few months ago and then unleashed this SteamVR bug. It only flips a few bytes in the SteamVR client so you can actually download the packages and there will be no game launches in the interim and you immediately switch off the Steam client after download anyway.

If you are paranoid, create a burner Steam account and login with it to download SteamVR (it's free after all to all users) and then download the older files. You can keep all the files you download afterward so you never have to run the Cheat Engine. 

Honestly, someone should just upload the whole 1.10.26 SteamVR payload to a file sharing service and link it back on Valve's official page so they actually get them to fix it.

**Comment by mirage335** [comment]
There is a way to just copy/paste the current installation of SteamVR to run standalone later?

**Comment by Da-Nuker** [comment]
Once you downloaded the files, you can keep that copy in a spare folder anywhere on your computer and copy it back into your steam\common folders (and replace the appmanifest_250820.acf) file. It is also highly recommended as you will need definitely need to keep a spare copy of the SteamVR folders because if you launch SteamVR directly, it will run the update and you need to replace all the files again.

**Comment by black-light-studio** [comment]
I've spent over 1K€ trying to get a good frame-rate, damn you SteamVR ...
Samsung Odyssey+ only getting 20-25 FPS ... 
running a 10900K at 5.3GHz with a 1080TI
FML time to see if I can somehow route it trough a capture card

**Comment by BOLL7708** [comment]
@black-light-studio Just as a side note, if you use a capture card to intercept the headset video signal you will most likely get an image with distortion applied. Meaning chromatic aberration and barrel distortion. 

If the game provides a good game window that can obviously be used instead, but if you're like me and record a plethora of games where some only provide a tiny output window, it's a no go 😅
